### DISCUSSION TITLE
How do start image processing in python?

### DISCUSSION THREAD

**Discussion by Abdol1381** [community/community#143155] [Programming Help] [answered]
### Select Topic Area

Show & Tell

### Body

_import Required libraries_

to get start with image processing in python follow these key steps:

The main library used for image processing in python is OpenCV:

`import cv2`
`import numpy as np`
`from skimage.io import imread `
`import matplotlib.pyplot as plt`

load the image 

Use imread() from sckit-image or cv2.imred from OpenCV to load image into python

img = imread('image.jpg')

**Comment by nabinghosh** [comment, marked as answer]
# 🖼️ Image Processing in Python - Getting Started Guide

Hey fellow developers! 👋

I wanted to share a beginner-friendly guide to get started with image processing in Python. Here's everything you need to know:

## 📚 First, install the required libraries:
```bash
pip install opencv-python
pip install pillow
pip install numpy

```
💻 Here's a basic starter code:
```
import cv2
import numpy as np
from PIL import Image

# Basic OpenCV operations
def basic_image_processing():
    # Read an image 
    img = cv2.imread('image.jpg')
    
    # Common operations 
    resized = cv2.resize(img, (300, 300))
    gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
    blurred = cv2.GaussianBlur(img, (5, 5), 0)
    
    # Save processed image
    cv2.imwrite('processed_image.jpg', blurred)

# Using PIL for simple operations
def pil_operations():
    # Open image
    img = Image.open('image.jpg')
    
    # Basic transformations
    resized = img.resize((300, 300))
    gray = img.convert('L')
    rotated = img.rotate(45)
    
    # Save result
    rotated.save('rotated_image.jpg')
```
🔑Key Things to Remember:

- OpenCV uses BGR color format, while PIL uses RGB

- Always check if images are loaded successfully

- Handle exceptions for missing files

- Be careful with memory when processing large images
